### PR TITLE
chore: Update IPLD-adjacent dependencies

### PR DESCRIPTION
**Pull request by cdata**
BREAKING CHANGE: Several critical dependencies of this library were updated to new versions that contain breaking changes.

### CODE REVIEW

File: rust/noosphere-ns/src/records.rs [hunk 1]
```diff
@@ -202,15 +202,15 @@ impl FromStr for NSRecord {
         // Wait for next release of `ucan` which includes traits and
         // removes `try_from_token_string`:
         // https://github.com/ucan-wg/rs-ucan/commit/75e9afdb9da60c3d5d8c65b6704e412f0ef8189b
-        Ok(NSRecord::new(Ucan::try_from_token_string(ucan_token)?))
```
Comment (jsantell) [inline rust/noosphere-ns/src/records.rs:205]: Can remove this comment now with the new UCAN

**Review by jsantell — APPROVED** [review]
LGTM! There's an extra comment/todo about the UCAN update in `NSRecord` that might be worth taking a look

File: rust/noosphere-ns/src/records.rs (hunk 1 above)
Comment (jsantell) [reply rust/noosphere-ns/src/records.rs:205]: Might be some other changes via the addition of traits in ucan package, I can take a look separately

**Review by jsantell — APPROVED** [review]

File: rust/noosphere-ns/src/records.rs (hunk 1 above)
Comment (cdata) [reply rust/noosphere-ns/src/records.rs:205]: If it's alright, I'll let you take care of this one for me :pray:

**Pull request merged**